### PR TITLE
Add workflow to handle long running issues

### PR DESCRIPTION
**Pull request by jvschneid**
*Description of changes:* Add workflow to handle long running issues and auto-close given labels and time windows.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/stale-issues.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+          # If you don't want to mark a issue as being ancient based on a
+          # threshold of "upvotes", you can set this here. An "upvote" is
+          # the total number of +1, heart, hooray, and rocket reactions
+          # on an issue.
+          minimum-upvotes-to-exempt: 5
+
+          # Don't set closed-for-staleness label to skip closing very old issues
+          # regardless of label
+          closed-for-staleness-label: closed-for-staleness
```
Comment (kstich) [inline .github/workflows/stale-issues.yml:40]: This label doesn't exist.

File: .github/workflows/stale-issues.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    name: Stale issue job
+    steps:
+      - uses: aws-actions/stale-issue-cleanup@v3
+        with:
+          # Targeted types
+          issue-types: issues
+
+          # Issue timing
+          days-before-stale: 60
+          days-before-close: 7
```
Comment (kstich) [inline .github/workflows/stale-issues.yml:20]: Let's up this to 14 days.

**Review by kstich — CHANGES_REQUESTED** [review]

File: .github/workflows/stale-issues.yml (hunk 1 above)
Comment (jvschneid) [reply .github/workflows/stale-issues.yml:40]: Yes, my plan is to create it once the PR is approved and there are no concerns with naming (:

File: .github/workflows/stale-issues.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          issue-types: issues
+
+          # Issue timing
+          days-before-stale: 60
+          days-before-close: 14
+          days-before-ancient: 365
+
+          # Labels
+          stale-issue-label: closing-soon
+          exempt-issue-label: documentation,feature-request
```
Comment (kstich) [inline .github/workflows/stale-issues.yml:25]: Let's look at the labels on the repo/in-use and see if this is the right list. It probably needs at least `bug` added.

**Comment by kstich** [comment]
Should also make sure all the issues in the repo are properly labeled before merging this, so it doesn't start auto-closing things we previously didn't label properly.

**Comment by jvschneid** [comment]
> Should also make sure all the issues in the repo are properly labeled before merging this, so it doesn't start auto-closing things we previously didn't label properly.

I did a pass on the issues this morning, feel free to do a quick glance at the unlabeled ones and see if I missed any that should be labeled though (:

File: .github/workflows/stale-issues.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          issue-types: issues
+
+          # Issue timing
+          days-before-stale: 60
+          days-before-close: 14
+          days-before-ancient: 365
+
+          # Labels
+          stale-issue-label: closing-soon
+          exempt-issue-label: bug,documentation,feature-request,guidance,protocol-test
```
Comment (kstich) [inline .github/workflows/stale-issues.yml:25]: Shouldn't close `investigating`.

I think `guidance` will need to wait to come off this list until we've either tagged them as `documentation` or chosen that they need no further update.

Comment (kstich) [reply .github/workflows/stale-issues.yml]: I've labeled everything with a `guidance` label that needs more work with another label representing that work. The `guidance` label can be removed from this list now and we can move forward.

**Review by kstich — APPROVED** [review]

**Pull request merged**